assets/js/steps/check-in-items.js

```diff
@@ -73,59 +128,75 @@ function CheckInItems({ actionHandler }) {
         };
     }, [actionHandler]);
 
-    let items = [];
-    if (context.machineState.get.items) {
-        items = adaptListOfBooksToBanner(context.machineState.get.items);
-    }
-
     /**
-     * Handles numpad presses.
-     *
-     * @param key
-     *    The pressed button.
+     * Clear new reservation.
      */
-    function onNumPadPress(key) {
-        const typedBarcode = `${scannedBarcode}`;
-        setActiveBanner(false);
-        switch (key) {
-            case deleteButtonLabel:
-                setScannedBarcode(typedBarcode.slice(0, -1));
-                break;
-            case okButtonLabel:
-                setActiveBanner(true);
-                handleItemCheckIn(scannedBarcode);
-                break;
-            default:
-                setScannedBarcode(`${typedBarcode}${key}`);
-                break;
-        }
-    }
+    useEffect(() => {
+        setNewReservation(null);
+    }, [newReservation]);
 
     /**
-     * Handles keyboard inputs.
-     *
-     * @param target
-     *    The pressed target.
+     * Determines whether to play a soumd and which to play.
      */
-    function onKeyboardInput({ target }) {
-        setActiveBanner(false);
-        setScannedBarcode(target.value);
-    }
+    useEffect(() => {
+        /**
+        * Evaluate if a new checked-in book is reserved by another user.
+        */
+        if (context.machineState.get.items === undefined) return;
 
-    /**
-     * Handles keyboard inputs.
-     *
-     */
-    function handleItemCheckIn() {
-        setActiveBanner(true);
-        actionHandler('checkInItem', {
-            itemIdentifier: scannedBarcode
+        let playSound = false;
```

Comment:
It's 2 variables to see if and what sound should be played. It could be reduced to 1.

Remove this:
```
let playSound = false;
let soundToPlay = '';
```

Replace with
```
let playSound = null;
```
...
```
if (context.boxConfig.get.soundEnabled && playSound !== null) {
    sound.playSound(playSound);
}
```